tools/autograd/derivatives.yaml

```diff
@@ -1250,6 +1250,20 @@
 
 - name: slice_backward(Tensor grad_output, int[] input_sizes, int dim, int start, int end, int step) -> Tensor
   grad_output: grad.slice(dim, start, end, step)
+
+- name: slice_scatter(Tensor self, Tensor src, int dim=0, int? start=None, int? end=None, int step=1) -> Tensor
+  self: slice_scatter(grad, at::zeros(src.sizes(), src.options()), dim, start, end, step)
```

Comment:
cc @zou3519, src.options police! @albanD I wonder if we could get away with using the mutable version of this in the formula here, saves the second copy

Reply:
I would actually expect that src would broadcast here and so you can do `at::zeros({}, ...)` to avoid this problem.

Also it is *never* allowed to modify `grad` inplace!

Reply:
This callsite isn't a problem for vmap because slice_scatter is an out-of-place function.

But could we rewrite `at::zeros(src.sizes(), src.options())` as `torch::zeros_like(srcs)`? That's pretty much what zeros_like does and it has the added benefit of being less verbose